scripts/src/admin/translations.ts

```diff
@@ -150,8 +158,136 @@ export async function getCsvTemplateData() {
+poll,offer_contents:OKKmC7OHkK2GztnaD3VF3:content:pages[1].question.answers[1],Daffodil,Narciso,22
+poll,offer_contents:OKKmC7OHkK2GztnaD3VF3:content:pages[1].question.answers[2],Petunia,Petúnia,22
+poll,offer_contents:OKKmC7OHkK2GztnaD3VF3:content:pages[1].question.answers[3],Daisy,Margarida,22
+ ***/
+
+export type CsvParse = ((input: Buffer, options?: Options) => any) & typeof csvParse;
+
+type TranslationDataRow = [string, string, string, string, number];
+type TranslationData = TranslationDataRow[];
+type OfferTranslationData = {
+	title: string,
+	description: string,
+	orderDescription: string,
+	orderTitle: string,
+	content: any /* It's suppose to be offerContent: { content } but that drives TS crazy */
+};
+type Column = "title" | "description" | "orderDescription" | "orderTitle" | "content";
+type Table = "offer" | "offerContent";
+type OffersTranslation = { [index: string]: OfferTranslationData };
+type OffersTranslationRow = {
+	offer: Offer;
+	offerId: string;
+	context: "offer" | "offer_content";
+	path: string;
+	language: string;
+	translation: string;
+};
+
+function getCsvKeyElements(key: string): [Table, string, Column, string] {
+	return key.split(":") as [Table, string, Column, string];
+}
+
+function getOfferContentFromJson(offerContent?: OfferContent) {
+	if (!offerContent) {
+		return {};
+	}
+	return parseContent(offerContent.content);
+}
+
+async function insertIntoDb(data: OffersTranslation, language: string) {
+	const allOffers = await Offer.find({ type: "earn" });
+	const dbReadyData: OffersTranslationRow[] = [];
+	Object.entries(data).forEach(offerData => {
+		const [offerId, offerTranslations] = offerData;
+		const offer = allOffers.find(offer => offer.id === offerId);
+		if (!offer) {
+			console.warn("DB missing offer, offer ID:", offerId);
+			return;
+		}
+		Object.entries(offerTranslations).forEach(offerTranslationData => {
+			const [column, translation] = offerTranslationData as [string, string];
+			if (column === "content") {
+				dbReadyData.push({
+					context: "offer_content",
+					translation: JSON.stringify(translation),
+					offer,
+					path: column,
+					language,
+					offerId: offer.id,
+				});
+			} else {
+				dbReadyData.push({
+					context: "offer",
+					translation,
+					offer,
+					path: column,
+					language,
+					offerId: offer.id,
+				});
+			}
+		});
+	});
+	OfferTranslation.createQueryBuilder().insert().values(dbReadyData).execute();
+}
+
+//  TODO: add validation
+async function processTranslationData(csvData: TranslationData) {
+	const allOfferContents = await OfferContent.find({ select: ["offerId", "content"] } as FindManyOptions<OfferContent>);
+	const allContentTranslations: OffersTranslation = {};
+	csvData.forEach(row => {
+		const [__, csvKey, ___, translation] = row;
+		const [table, offerId, column, jsonPath] = getCsvKeyElements(csvKey);
+		const offerTranslations = (offerId in allContentTranslations) ?
```

Comment:
didn't we say that you're breaking this to a more readable `if () { } else {}` block?